app/src/chat_api.py

```diff
@@ -197,15 +198,33 @@ class Citation(BaseModel):
     headings: Sequence[str]
     citation_text: str
 
+    @staticmethod
+    def format_highlighted_uri(source_url: str | None, subsection_text: str) -> str | None:
+        if not source_url:
+            return None
+        citation_without_special_chars = re.sub(r"\W+", " ", subsection_text).strip()
+        citation_arr = citation_without_special_chars.split(" ")
+        formatted_text_to_highlight = (
+            "%20".join(citation_arr[:5]) + "," + "%20".join(citation_arr[-5:])
+        )
+        return (
+            (source_url[:-1] if source_url.endswith("/") else source_url)
+            + "#:~:text="
+            + formatted_text_to_highlight
+        )
```

Comment:
I might pull this out into a free-standing function since it's not specific to the Citation class (e.g., this could go in a generic utilities file)